airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/ctid/PostgresCtidHandler.java

```diff
@@ -110,47 +123,100 @@ public List<AutoCloseableIterator<AirbyteMessage>> getIncrementalIterators(
     return iteratorList;
   }
 
+  /**
+   * Builds a plan for subqueries that will return an approximate size of data.
+   * Using information about a table size and block (page) size.
+   *
+   * @param startCtid starting point
+   * @param relationSize table size
+   * @param blockSize page size
+   * @param chunkSizeGB required amount of data in each partition
+   * @return a list of ctid that can be used to generate queries.
+   */
+  @VisibleForTesting
+  static List<Pair<Ctid, Ctid>> ctidQueryPlan(final Ctid startCtid, final long relationSize, final long blockSize, final int chunkSizeGB) {
+    final List<Pair<Ctid, Ctid>> chunks = new ArrayList<>();
+    long lowerBound = startCtid.page;
+    long upperBound;
+    final double oneGigaPages = MEGABYTE * 1000 / blockSize;
```

Comment:
nit : can just have one GB

Reply:
didn't get that

Reply:
Was just saying we can have : 
final double oneGigaPages = ONE_GIGABYTE_BYTES / blockSize; instead